### PR TITLE
Add more "gotcha" tips

### PR DESCRIPTION
**Pull request by rvl**
Here is some documentation of things I have run into when using iohk-nix/Haskell.nix.

I could move some of this into the Haskell.nix docs some time, but in the meantime it's captured here.

### CODE REVIEW

File: gotchas/gotcha-2.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+```
+
+This means that the resolver specified in `stack.yaml` was not found
+in `stackage.nix`. The snapshot is probably more recent than your
+version of `stackage.nix`.
+
+The repos `stackage.nix` and `hackage.nix` are updated daily, along
+with their JSON pins in `haskell.nix`.
+
+The best course of action is usually to update your `haskell.nix`
```
Comment (angerman) [inline gotchas/gotcha-2.md]: should we provide an example here?

File: gotchas/gotcha-2.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+collection that you are using. Nixpkgs will only keep one compiler of
+each major version.
+
+You may need to use a different Nixpkgs channel.
+
+If using `stack-to-nix`, you could choose a different Stackage snapshot.
+
+If using `plan-to-nix`, you could regenerate your plan and nix with a
+compiler version that exists in your Nixpkgs.
+
```
Comment (angerman) [inline gotchas/gotcha-2.md]: If you use `iohk-nix`/`input-output-hk/nixpkgs`, bumping that might help as well.

**Comment by domenkozar** [comment]
Missing gotcha: `pq` dependency missing -> should be added to https://github.com/input-output-hk/haskell.nix/blob/master/lib/system-nixpkgs-map.nix

**Comment by angerman** [comment]
> Missing gotcha: `pq` dependency missing -> should be added to https://github.com/input-output-hk/haskell.nix/blob/master/lib/system-nixpkgs-map.nix

Can you open a PR with the necessary change against `haskell.nix`?

**Pull request merged**